src/plugins/ini/ini.c

```diff
@@ -488,7 +459,41 @@ static int iniCommentToMeta (void * vhandle, const char * comment)
 {
 	CallbackHandle * handle = (CallbackHandle *)vhandle;
 	if (!handle->collectedComment) handle->collectedComment = keyNew ("/comments", KEY_CASCADING_NAME, KEY_END);
-	elektraMetaArrayAdd (handle->collectedComment, "comments", comment);
+	if (strncmp (comment, "#@META ", 7))
+		elektraMetaArrayAdd (handle->collectedComment, "comments", comment);
+	else
+	{
+		char * localCopy = elektraStrDup (comment + 7);
+		size_t len = strlen (localCopy);
+		char * ptr = localCopy;
+		char * name = ptr;
+		while (isspace (*name))
+			++name;
+		char * value;
+		ptr = strstr (localCopy, "=");
```

Comment:
Please give some docu what you are trying to do here. Refactoring might be useful, too.